components/filters.py

```diff
@@ -37,5 +37,6 @@ def filter_queryset(self, request, queryset, view):
 
         return (
                 get_objects_for_user(user, permission, queryset, accept_global_perms=False)
-                | queryset.filter(status=Component.Status.PUBLIC)
+                .union(queryset.filter(status=Component.Status.PUBLIC))
+                .order_by("pk")
```

Comment:
It's necessary to provide the `order_by` here since `union` doesn't preserve any `order_by` in the individual query sets.